code/game/machinery/iv_drip.dm

```diff
@@ -87,53 +87,56 @@
 
 
 /obj/machinery/iv_drip/process()
-	if(src.attached)
+	if(!attached)
+		return
 
-		if(!(get_dist(src, src.attached) <= 1 && isturf(src.attached.loc)))
-			visible_message("The needle is ripped out of [src.attached], doesn't that hurt?")
-			attached.apply_damage(3, BRUTE, pick("r_arm", "l_arm"))
-			attached = null
+	if(!(get_dist(src, attached) <= 1 && isturf(attached.loc)))
+		visible_message("The needle is ripped out of [attached], doesn't that hurt?")
+		attached.apply_damage(3, BRUTE, pick("r_arm", "l_arm"))
+		attached = null
+		update_icon()
+		STOP_PROCESSING(SSobj, src)
+		return
+
+	if(!beaker)
+		return
+
+	// Give blood
+	if(mode)
+		if(beaker.volume > 0)
+			var/transfer_amount = REAGENTS_METABOLISM
+			if(istype(src.beaker, /obj/item/reagent_container/blood))
+				// speed up transfer on blood packs
+				transfer_amount = 4
+			attached.inject_blood(beaker, transfer_amount)
 			update_icon()
-			stop_processing()
-			return
 
-	if(attached && beaker)
-		// Give blood
-		if(mode)
-			if(beaker.volume > 0)
-				var/transfer_amount = REAGENTS_METABOLISM
-				if(istype(src.beaker, /obj/item/reagent_container/blood))
-					// speed up transfer on blood packs
-					transfer_amount = 4
-				attached.inject_blood(beaker, transfer_amount)
-				update_icon()
-
-		// Take blood
-		else
-			var/amount = beaker.reagents.maximum_volume - beaker.reagents.total_volume
-			amount = min(amount, 4)
-			// If the beaker is full, ping
-			if(amount == 0)
-				if(prob(5)) visible_message("\The [src] pings.")
-				return
+	// Take blood
+	else
+		var/amount = beaker.reagents.maximum_volume - beaker.reagents.total_volume
+		amount = min(amount, 4)
+		// If the beaker is full, ping
+		if(amount == 0)
+			if(prob(5)) visible_message("\The [src] pings.")
+			return
 
-			var/mob/living/carbon/human/T = attached
+		var/mob/living/carbon/human/T = attached
 
-			if(!istype(T)) return
-			if(!T.dna)
-				return
-			if(NOCLONE in T.mutations)
-				return
+		if(!istype(T)) return
+		if(!T.dna)
+			return
+		if(NOCLONE in T.mutations)
+			return
 
-			if(T.species && T.species.species_flags & NO_BLOOD)
-				return
+		if(T?.species.species_flags & NO_BLOOD)
```

Comment:
```suggestion
		if(T.species?.species_flags & NO_BLOOD)
```
We checked for `T` in the `istype(T)` a few lines above.

Reply:
You're right this was misplaced, thanks.